# Playstore Data analysis

In this  project i am going to analyzed the google playstore dataset using python pandas DataFrame . The data is available in kaggle in csv format.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv(r"C:\Users\utsha\OneDrive\Desktop\Data analysis project\dataset\9. googleplaystore.csv")

In [3]:
df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [4]:
df.shape  # checking rows and columns size of data 

(10841, 13)

### Handling missing values

In [5]:
df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

There are some missing values,
* rating missing values can be replace using average rating
* current ver,android ver,content rating is object data type so it is replace with highest frequenct data


In [6]:
df_backup = df.copy()  # creating backup

In [7]:
import numpy as np

In [8]:
df['Rating'].replace(np.nan , df['Rating'].mean(),inplace = True)  # replacing null values in rating by average rating 

In [9]:
df.Rating.isnull().sum()

0

In [10]:
df['Current Ver'].value_counts()

Varies with device    1459
1.0                    809
1.1                    264
1.2                    178
2.0                    151
                      ... 
1.0.17.3905              1
15.1.2                   1
4.94.19                  1
1.1.11.11                1
2.0.148.0                1
Name: Current Ver, Length: 2832, dtype: int64

In [11]:
df['Android Ver'].value_counts()

4.1 and up            2451
4.0.3 and up          1501
4.0 and up            1375
Varies with device    1362
4.4 and up             980
2.3 and up             652
5.0 and up             601
4.2 and up             394
2.3.3 and up           281
2.2 and up             244
4.3 and up             243
3.0 and up             241
2.1 and up             134
1.6 and up             116
6.0 and up              60
7.0 and up              42
3.2 and up              36
2.0 and up              32
5.1 and up              24
1.5 and up              20
4.4W and up             12
3.1 and up              10
2.0.1 and up             7
8.0 and up               6
7.1 and up               3
4.0.3 - 7.1.1            2
5.0 - 8.0                2
1.0 and up               2
7.0 - 7.1.1              1
4.1 - 7.1.1              1
5.0 - 6.0                1
2.2 - 7.1.1              1
5.0 - 7.1.1              1
Name: Android Ver, dtype: int64

In [12]:
df['Content Rating'].value_counts()

Everyone           8714
Teen               1208
Mature 17+          499
Everyone 10+        414
Adults only 18+       3
Unrated               2
Name: Content Rating, dtype: int64

In [13]:
df['Current Ver'].replace(np.nan , 'Varies with device',inplace = True) # replaceing categorical data with highest frequency
df['Android Ver'].replace(np.nan , '4.1 and up',inplace = True)
df['Content Rating'].replace(np.nan , 'Everyone',inplace = True)

In [14]:
df.isnull().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              1
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       0
Android Ver       0
dtype: int64

In [15]:
df.dropna(inplace = True)

In [16]:
df.isnull().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       0
Android Ver       0
dtype: int64

In [17]:
df.shape

(10840, 13)

we have now cleaned data frame


## Formatting of data

In [18]:
df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [19]:
df.dtypes   

App                object
Category           object
Rating            float64
Reviews            object
Size               object
Installs           object
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

In [41]:
#df['Reviews'] = df['Reviews'].str.replace('3.0M','3.0')
df['Reviews'] = df['Reviews'].astype('float')

## Now its time for some valuable insights from data

In [42]:
df.describe(include = 'all')

App Category        Rating       Reviews                Size  \
count    10840    10840  10840.000000  1.084000e+04               10840   
unique    9659       34           NaN           NaN                 462   
top     ROBLOX   FAMILY           NaN           NaN  Varies with device   
freq         9     1971           NaN           NaN                1694   
mean       NaN      NaN      4.193338  4.441529e+05                 NaN   
std        NaN      NaN      0.499580  2.927761e+06                 NaN   
min        NaN      NaN      1.000000  0.000000e+00                 NaN   
25%        NaN      NaN      4.100000  3.800000e+01                 NaN   
50%        NaN      NaN      4.200000  2.094000e+03                 NaN   
75%        NaN      NaN      4.500000  5.477550e+04                 NaN   
max        NaN      NaN     19.000000  7.815831e+07                 NaN   

          Installs   Type  Price Content Rating Genres    Last Updated  \
count        10840  10840  10840          10840  10840           10840   
unique          21      3     93              6    120            1378   
top     1,000,000+   Free      0       Everyone  Tools  August 3, 2018   
freq          1579  10039  10039           8715    842             326   
mean           NaN    NaN    NaN            NaN    NaN             NaN   
std            NaN    NaN    NaN            NaN    NaN             NaN   
min            NaN    NaN    NaN            NaN    NaN             NaN   
25%            NaN    NaN    NaN            NaN    NaN             NaN   
50%            NaN    NaN    NaN            NaN    NaN             NaN   
75%            NaN    NaN    NaN            NaN    NaN             NaN   
max            NaN    NaN    NaN            NaN    NaN             NaN   

               Current Ver Android Ver  
count                10840       10840  
unique                2832          33  
top     Varies with device  4.1 and up  
freq                  1466        2454  
mean                   NaN         NaN  
std                    NaN         NaN  
min                    NaN         NaN  
25%                    NaN         NaN  
50%                    NaN         NaN  
75%                    NaN         NaN  
max                    NaN         NaN

# Total numbes of app contains Astrology in title

In [43]:
df.head(2)

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                             Coloring book moana  ART_AND_DESIGN     3.9   

   Reviews Size  Installs  Type Price Content Rating  \
0    159.0  19M   10,000+  Free     0       Everyone   
1    967.0  14M  500,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up

In [44]:
df[df['App'].str.contains('Astrology')]

App   Category  Rating  \
1570   Horoscopes – Daily Zodiac Horoscope and Astrology  LIFESTYLE     4.6   
1592              သိင်္ Astrology - Min Thein Kha BayDin  LIFESTYLE     4.7   
10840      iHoroscope - 2018 Daily Horoscope & Astrology  LIFESTYLE     4.5   

        Reviews Size     Installs  Type Price Content Rating     Genres  \
1570   161143.0  11M  10,000,000+  Free     0   Everyone 10+  Lifestyle   
1592     2225.0  15M     100,000+  Free     0       Everyone  Lifestyle   
10840  398307.0  19M  10,000,000+  Free     0       Everyone  Lifestyle   

        Last Updated         Current Ver         Android Ver  
1570   June 25, 2018          5.2.4(881)        4.0.3 and up  
1592   July 26, 2018               4.2.1        4.0.3 and up  
10840  July 25, 2018  Varies with device  Varies with device

In [45]:
len(df[df['App'].str.contains('Astrology')])

3

3 app contains astrology in app titles

# Checking average app rating in playstore

In [46]:
df.Rating.mean()

4.1933383153622925

The average app rating in playstore is 4.1919246413700515

# Checking which category gets the highest rating

In [47]:
df.head(2)

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                             Coloring book moana  ART_AND_DESIGN     3.9   

   Reviews Size  Installs  Type Price Content Rating  \
0    159.0  19M   10,000+  Free     0       Everyone   
1    967.0  14M  500,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up

In [48]:
df.groupby('Category')['Rating'].mean().sort_values(ascending = False)

Category
1.9                    19.000000
EDUCATION               4.387778
EVENTS                  4.363647
ART_AND_DESIGN          4.350462
BOOKS_AND_REFERENCE     4.311026
PERSONALIZATION         4.307603
GAME                    4.282506
PARENTING               4.282223
HEALTH_AND_FITNESS      4.266296
BEAUTY                  4.260882
SHOPPING                4.254052
SOCIAL                  4.248001
WEATHER                 4.239675
SPORTS                  4.218404
PRODUCTIVITY            4.208287
HOUSE_AND_HOME          4.196819
FAMILY                  4.192394
PHOTOGRAPHY             4.192179
AUTO_AND_VEHICLES       4.190824
MEDICAL                 4.190167
LIBRARIES_AND_DEMO      4.181962
FOOD_AND_DRINK          4.170709
COMMUNICATION           4.163842
COMICS                  4.156445
BUSINESS                4.145987
NEWS_AND_MAGAZINES      4.142993
FINANCE                 4.139108
ENTERTAINMENT           4.126174
TRAVEL_AND_LOCAL        4.119716
LIFESTYLE               4.112427
V

Education has highest rating and dating has the lowest rating


# Totals number of apps having five star rating

In [49]:
df[df.Rating == 5].Rating.value_counts()

5.0    274
Name: Rating, dtype: int64

274 app have 5 stars rating

# Average value of reviews

In [50]:
df.head(2)

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                             Coloring book moana  ART_AND_DESIGN     3.9   

   Reviews Size  Installs  Type Price Content Rating  \
0    159.0  19M   10,000+  Free     0       Everyone   
1    967.0  14M  500,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up

In [51]:
df.Reviews.mean()

444152.8963099631

Around 444152 peoples make review in playstore per app in average

# Total number of free and paid apps 



In [52]:
df.head(2)

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                             Coloring book moana  ART_AND_DESIGN     3.9   

   Reviews Size  Installs  Type Price Content Rating  \
0    159.0  19M   10,000+  Free     0       Everyone   
1    967.0  14M  500,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up

In [53]:
df[df.Type == 'Free'].Type.value_counts()

Free    10039
Name: Type, dtype: int64

In [54]:
df[df.Type == 'Paid'].Type.value_counts()

Paid    800
Name: Type, dtype: int64

* Free apps = 10038
* Paid apps = 799

In [55]:
df[df.Reviews == df.Reviews.max()]

App Category  Rating     Reviews                Size  \
2544  Facebook   SOCIAL     4.1  78158306.0  Varies with device   

            Installs  Type Price Content Rating  Genres    Last Updated  \
2544  1,000,000,000+  Free     0           Teen  Social  August 3, 2018   

             Current Ver         Android Ver  
2544  Varies with device  Varies with device

In [56]:
index = df['Reviews'].sort_values(ascending = False).head().index
df.iloc[index]['App'].to_frame()

App
2544            Facebook
3943            Facebook
381   WhatsApp Messenger
336   WhatsApp Messenger
3904  WhatsApp Messenger

# Average rating of free and paid apps


In [57]:
df.groupby('Type')['Rating'].mean()

Type
0       19.000000
Free     4.187141
Paid     4.252601
Name: Rating, dtype: float64

We can see that paid got some better reviews than free apps


# Top 5 most installed apps

In [76]:
df.head(2)

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                             Coloring book moana  ART_AND_DESIGN     3.9   

   Reviews Size  Installs  Type Price Content Rating  \
0    159.0  19M   10,000+  Free     0       Everyone   
1    967.0  14M  500,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up

In [85]:
#df['Installs'] = df['Installs'].str.replace('Free','0')

In [90]:
index = df.sort_values(by='Installs',ascending = False).head().index
df.iloc[index]['App'].to_frame()

App
1661                           Temple Run 2
4039  Google Duo - High Quality Video Calls
385                         Viber Messenger
3562                        Google Calendar
4048                                Dropbox